### PR TITLE
Rename methods and parameters for consistency across SDKs

### LINKED ISSUE #409: Rename methods and parameters for consistency across SDKs
## Expected Behavior

Java, .Net, Go and Python SDK have consistent method and parameter names.

Feature | Type | Current | New |
-|-|-|-
Service Invocation | method | invokeService | invokeMethod
Service Invocation | parameter | method | methodName
Service Invocation | parameter | request | data
State Store | method | saveStates | saveBulkState
State Store | method | getStates | getBulkState
State Store | method | executeTransaction | executeStateTransaction
State Store | parameter | stateStoreName | storeName
Secret Store | parameter | secretStoreName | storeName
Binding | parameter | name | bindingName
PubSub | parameter | topic | topicName
Actor | method | invokeActorMethod | invoke
Actor | method | registerActorTimer | registerTimer

## Actual Behavior

Java, .Net, Go and Python SDK have inconsistent method and parameter names.

## Steps to Reproduce the Problem

Use any of the methods in different SDKs to notice the difference in naming.

## Release Note

RELEASE NOTE: **RENAME** methods and properties for consistency across SDKs.

Parent: https://github.com/dapr/dapr/issues/2535

### PR DESCRIPTION
**Pull request by arghya88**
Signed-off-by: Arghya Sadhu <arghya88@gmail.com>

# Description

Rename methods and parameters for consistency across SDKs

## Issue reference

Fixes #409 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: sdk/src/main/java/io/dapr/client/AbstractDaprClient.java
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/AbstractDaprClient.java:337]: naming inconsistency.
`saveBulkState`

File: sdk/src/main/java/io/dapr/client/AbstractDaprClient.java
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/AbstractDaprClient.java:348]: do not rename this. this will remain `saveState` similar to the `getState` counterparts.

File: sdk/src/main/java/io/dapr/client/DaprClient.java
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/DaprClient.java:113]:
```suggestion
   * Invoke a service method, using serialization.
```
And similar one for all the invokeMethod variants.

File: sdk/src/main/java/io/dapr/client/DaprClient.java
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/DaprClient.java:412]: @artursouza should the domain class `GetStatesRequest` also be renamed to `GetBulkStateRequest` ?

File: sdk/src/main/java/io/dapr/client/DaprClient.java
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/DaprClient.java:438]: `saveBulkState`

File: sdk/src/main/java/io/dapr/client/DaprClient.java
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/DaprClient.java:446]: `saveBulkState`

File: sdk/src/main/java/io/dapr/client/DaprClient.java
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/DaprClient.java:456]: `saveState` no change

File: sdk/src/main/java/io/dapr/client/DaprClient.java
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/DaprClient.java:468]: `saveState` no change.

**Review by mukundansundar — COMMENTED** [review]
@arghya88 Thanks for your contribution. I have left some comments in the PR.

File: sdk-actors/src/main/java/io/dapr/actors/runtime/DaprClient.java
Comment (artursouza) [reply sdk-actors/src/main/java/io/dapr/actors/runtime/DaprClient.java:71]: Yes, please remove "Actor" from all methods here.

File: sdk/src/main/java/io/dapr/client/AbstractDaprClient.java
Comment (artursouza) [reply sdk/src/main/java/io/dapr/client/AbstractDaprClient.java:348]: Correct. This is singular.

File: sdk/src/main/java/io/dapr/client/DaprClient.java
Comment (artursouza) [reply sdk/src/main/java/io/dapr/client/DaprClient.java:412]: Please, rename request classes too.

**Comment by artursouza** [comment]
@arghya88 Thanks a lot for your contribution. The issue was not clear on every detail on the renaming, please see comments on the PR. We appreciate your help here.

**Comment by arghya88** [comment]
@artursouza @mukundansundar thanks for the feedback. Have incorporated the changes.PTAL.

**Comment by mukundansundar** [comment]
@arghya88 Please remove all html changes. The docs will be updated during the release of the sdk.

File: sdk/src/main/java/io/dapr/client/DaprClient.java
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/DaprClient.java:231]: nitpick: `The name of the binding to call.`

File: sdk/src/test/java/io/dapr/client/DaprClientGrpcTest.java
Comment (mukundansundar) [inline sdk/src/test/java/io/dapr/client/DaprClientGrpcTest.java:13]: ideally do not have `*` imports.

**Review by mukundansundar — COMMENTED** [review]
Rename `GetStatesRequestBuilder` also to `GetBulkStateRequestBuilder`
@arghya88 left a few more comments.

**Comment by arghya88** [comment]
> Rename `GetStatesRequestBuilder` also to `GetBulkStateRequestBuilder`
> @arghya88 left a few more comments.

@mukundansundar Done

File: sdk/src/main/java/io/dapr/client/domain/GetBulkStateRequestBuilder.java
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/domain/GetBulkStateRequestBuilder.java:19]: `storeName` for consistency?

File: sdk/src/main/java/io/dapr/client/domain/GetBulkStateRequest.java
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/domain/GetBulkStateRequest.java:17]: `storeName` for consistency?

File: sdk/src/main/java/io/dapr/client/DaprClient.java
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/DaprClient.java:242]: nitpick: `The name of the binding to call.` 
also all occurrences of this argument.

**Review by mukundansundar — COMMENTED** [review]
@arghya88 Just a few more changes. Please resolve conflicts.
Rename `stateStoreName` to `storeName` in `GetStateRequest`, `GetStateRequestBuilder`, `SaveStateRequest`, `SaveStateRequestBuilder`, `GetBulkState` and `GetBulkStateRequestBuilder`. 
Also change `secretStoreName` to `storeName` in `GetSecretRequest` and GetSecretRequestBuilder`

@artursouza any thoughts on these changes?

File: sdk/src/main/java/io/dapr/client/AbstractDaprClient.java
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/AbstractDaprClient.java:101]: @artursouza Should `body` be renamed to data?
Also should `InvokeServiceRequest` and associated builder be renamed to `InvokeMethodRequest` since the API methods are being renamed?

Comment (mukundansundar) [reply sdk/src/main/java/io/dapr/client/AbstractDaprClient.java:101]: @artursouza On second thought why is invokeService being renamed to invokeMethod? I think invokeService makes sense since we are invoking a particular method in that service.

Comment (artursouza) [reply sdk/src/main/java/io/dapr/client/AbstractDaprClient.java:101]: Let's discuss this with the maintainers and see if we can change it. Let's not merge this PR for this milestone, so we have time to change it.

**Comment by artursouza** [comment]
@arghya88 Please, address comments from @mukundansundar. Your PR is appreciated.

File: examples/src/main/java/io/dapr/examples/state/StateClient.java
Comment (artursouza) [inline examples/src/main/java/io/dapr/examples/state/StateClient.java:73]: Please, update README.md for all affected examples. We copy and paste code from here there.

File: sdk/src/main/java/io/dapr/client/AbstractDaprClient.java
Comment (arghya88) [reply sdk/src/main/java/io/dapr/client/AbstractDaprClient.java:101]: I would say let's do this in a followup PR after discussing with maintainers.

File: examples/src/main/java/io/dapr/examples/state/StateClient.java
Comment (arghya88) [reply examples/src/main/java/io/dapr/examples/state/StateClient.java:73]: Done

File: examples/src/main/java/io/dapr/examples/state/README.md
Comment (artursouza) [inline examples/src/main/java/io/dapr/examples/state/README.md:109]: getBulkState

**Review by artursouza — COMMENTED** [review]
Almost there.

File: sdk/src/main/java/io/dapr/client/AbstractDaprClient.java
Comment (artursouza) [reply sdk/src/main/java/io/dapr/client/AbstractDaprClient.java:101]: Well, we have cut the milestone already, so we can merge this PR soon and still have another PR to rename to invokeService again if maintainers agree to name it back.

File: examples/src/main/java/io/dapr/examples/state/README.md
Comment (arghya88) [reply examples/src/main/java/io/dapr/examples/state/README.md]: Done

**Comment by arghya88** [comment]
> Almost there.

@artursouza let me know if anything else is needed.

**Review by artursouza — APPROVED** [review]

**Pull request merged**
